components/brave_rewards/browser/rewards_p3a.cc

```diff
@@ -9,69 +9,38 @@
 #include "base/metrics/histogram_macros.h"
 #include "brave/components/brave_ads/common/pref_names.h"
 #include "brave/components/brave_rewards/common/pref_names.h"
+#include "brave/components/p3a_utils/bucket.h"
 #include "components/prefs/pref_service.h"
 
 namespace brave_rewards {
 namespace p3a {
 
-void RecordAutoContributionsState(AutoContributionsState state, int count) {
-  DCHECK_GE(count, 0);
-  int answer = 0;
-  switch (state) {
-    case AutoContributionsState::kNoWallet:
-      break;
-    case AutoContributionsState::kRewardsDisabled:
-      answer = 1;
-      break;
-    case AutoContributionsState::kWalletCreatedAutoContributeOff:
-      answer = 2;
-      break;
-    case AutoContributionsState::kAutoContributeOn:
-      switch (count) {
-        case 0:
-          answer = 3;
-          break;
-        case 1:
-          answer = 4;
-          break;
-        default:
-          answer = 5;
-      }
-      break;
-    default:
-      NOTREACHED();
-  }
-  UMA_HISTOGRAM_EXACT_LINEAR("Brave.Rewards.AutoContributionsState.2", answer,
-                             6);
+const char kTipsSentHistogramName[] = "Brave.Rewards.TipsSent";
+const char kAutoContributionsStateHistogramName[] =
+    "Brave.Rewards.AutoContributionsState.3";
+const char kAdsEnabledDurationHistogramName[] =
+    "Brave.Rewards.AdsEnabledDuration";
+const char kAdsStateHistogramName[] = "Brave.Rewards.AdsState.2";
+const int kTipsSentBuckets[] = {1, 3};
```

Comment:
Maybe `constexpr` for these?

Reply:
is there a benefit to making these compile time constants?

Reply:
I just try to use them when possible due to https://google.github.io/styleguide/cppguide.html#Use_of_constexpr